### PR TITLE
44069 Mental Health Clinic Display Issue

### LINKED ISSUE #44069: PROD AND STAGING:  Mental Health Clinic Display Issue:  Veterans w/No past MH Appts
Two preconditions for this issue to occur:
1.  User must navigate through a flow for another type of specialty care before selecting Mental Health
2.  User must NOT have any past Mental Health Appts within the last 36 mos.  
3.   Reviewed with dev team and this is a medium to low priority issue due to:  1) Vet will be dropped into request flow 2) This only affects vets without past MH appts 3) If Vet chose MH first and not another type of care this issue would not present 4) If vets refreshes pages or logs out and logs back in MH clinics would then displays.  

**Steps to to recreate:**
1. Login Select Type of care, select Mental Health, select Cheyenne. The following displays: Would you like to make an appointment at CHY MENTAL HEALTH ONE?(*Required). Yes, make my appointment here. No, I need a different clinic.
2. From the breadcrumbs navigate to Home Page.
3. Select type of care (other than Mental Health), if prompted Select VA facility. Select location. Several things could happen here: 1) Modal window displays with an explanation of why further action can't be taken OR 2) Vet dropped into Request flow OR 3) Vet dropped into Direct scheduling flow.
4. If modal window displays, close the modal and use the breadcrumbs to navigate back to home page. OR you can continue through the Request flow or Direct Scheduling flow OR you can use the breadcrumbs to navigate to Home page. What action you do at this point does NOT matter.
5. Navigate back on the Home page, select Mental Health.  Select Cheyenne.
6. System displays: Choose a clinic below or request a different clinic for this appointment.(*Required):  I need a different clinic
7. Expected Response:  Would you like to make an appointment at CHY MENTAL HEALTH ONE?(*Required). Yes, make my appointment here. No, I need a different clinic.
8. Actual Response:   Choose a clinic below or request a different clinic for this appointment.(*Required):  I need a different clinic

**To test this I used this patient:** john.morrison@id.me (for retesting you will need to verify that after I logged the defect that no one added MH appts for this patient.     I had Jenny McGibney temporarily move this patient to VAR-Resources.  

[https://app.zenhub.com/files/133843125/eca4ee5b-49f5-464a-9f40-f666a50ca167/download](https://app.zenhub.com/f iles/133843125/eca4ee5b-49f5-464a-9f40-f666a50ca167/download)

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
This PR fixes issue of clinic information not displaying when type of care is mental health.

**How to setup local environment**
* Configure 1 clinic for mental health appointments
  *  Modify ../service/mocks/index.js - 'GET /vaos/v0/facilities/:id/clinics' to return 1 clinic. i.e. return res.json({ data: [clinicList983.data[0]] });
* Remove appointments scheduled at clinic id = 308 (or just remove all appointments - easier)
  * ../services/mocks/var/past.json
  * ../services/mocks/var/confirmed_va.json
* Enable direct scheduling for that clinic
  * Change ../services/mocks/v2/scheduling_configurations.json:81 to true and line 87 to false
* Set feature flags to use v0 api
  * vaOnlineSchedulingVAOSServiceRequests = false
  * vaOnlineSchedulingVAOSServiceVAAppointments = false
  * vaOnlineSchedulingVAOSServiceCCAppointments = false
  * vaOnlineSchedulingStatusImprovement = false

See #44069 for steps to reproduce

**The error scenario occurs when:**
1st time through, type of care is mental health. Past appointments are not retrieved.
2nd time through, type of care is audiology. Past appointments are retrieved.
3rd time through, type of care is mental health. Past appointment information should not be used

Specifically, the error occurs when ../ClinicChoicePage/useClinicFormState/filteredClinics:69 is empty.

## Original issue(s)
fixes department-of-veterans-affairs/va.gov-team#44069

## Testing done
Unit testing

## Acceptance criteria
- [ ] All test should pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by jenniemc — CHANGES_REQUESTED** [review]
I used the Review Instance to test. The first attempt to make Mental Health VA appointment shows a single clinic available for direct schedule.
Screen Shot 2022-07-27 at 7 14 05 AM

The second attempt to make mental health appt no longer shows the clinic
Screen Shot 2022-07-27 at 7 20 00 AM

**Comment by vbahinwillit** [comment]
Updated instructions for testing locally.

**Review by jenniemc — CHANGES_REQUESTED** [review]
Looks good to me. Just one question and one suggestion.

File: src/applications/vaos/redux/selectors.js [hunk 1]
```diff
@@ -72,4 +72,4 @@ export const selectFeatureFilter36Vats = state =>
   toggleValues(state).vaOnlineFilter36Vats;
 
 export const selectFeatureVaosV2Next = state =>
-  toggleValues(state).vaOnlineSchedulingVaosV2Next;
+  toggleValues(state).vaOnlineSchedulingVaosv2next;
```
Comment (jenniemc) [inline src/applications/vaos/redux/selectors.js:75]: Why the name change? The flag does not seem to work locally unless we also replace the name on src/applications/vaos/services/mocks/index.js and src/platform/utilities/feature-toggles/featureFlagNames.js  in order for the flag to work on our local development. Luckily, this name change does not seem to impact vet-api toggle switch in staging environment, just our local environment. I suggest keeping the old name `vaOnlineSchedulingVaosV2Next`. 
```suggestion
  toggleValues(state).vaOnlineSchedulingVaosV2Next;
```

Comment (vbahinwillit) [reply src/applications/vaos/redux/selectors.js]: @jenniemc  - When using the review instance the vaOnlineSchedulingVaosV2Next was undefined. I don't know why but it seems to work with the change. I'm looking further into this.

Comment (vbahinwillit) [reply src/applications/vaos/redux/selectors.js]: Seems to be working now.

**Review by jenniemc — APPROVED** [review]
whoop whoop! This looks good to go!

File: src/applications/vaos/redux/selectors.js [hunk 2]
```diff
@@ -72,4 +72,4 @@ export const selectFeatureFilter36Vats = state =>
   toggleValues(state).vaOnlineFilter36Vats;
 
 export const selectFeatureVaosV2Next = state =>
-  toggleValues(state).vaOnlineSchedulingVaosV2Next;
+  toggleValues(state).vaOnlineSchedulingVAOSV2Next;
```
Comment (jenniemc) [inline src/applications/vaos/redux/selectors.js:75]: Great job catching the odd name conversion this feature toggle name.

**Review by theodur — APPROVED** [review]

**Pull request merged**